test-network-function/lifecycle/suite.go

```diff
@@ -334,21 +334,16 @@ func testPodsRecreation(env *config.TestEnvironment) {
 			env.ResetOc()
 			// drain node
 			drainNode(n.Name) // should go in this
+			// Ensure the node is uncordoned even if we have a failure in waitForAllDeploymentsReady
+			defer uncordonNode(n.Name)
```

Comment:
It gets a bit tricky to do defer inside a loop. Easier if the loop body becomes a function with a defer block.

Reply:
Ok, I think I got the idea. I'll add it in the next version.